api/pkg/observability/deployment/identity.go

```diff
@@ -0,0 +1,52 @@
+package deployment
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func enrichIdentityToPod(podSpec corev1.PodSpec, secretName string, containerNames []string) corev1.PodSpec {
```

Comment:
nit: Since this helper function only works by enriching pods with mounted Google application credentials stored in k8s secrets, can we mention this in the name of the function in some way? 😅 e.g. `enrichPodWithGoogleApplicationCredentials`, `mountGoogleApplicationCredentialsToPod` ...

Reply:
ah previously I planned to use an interface that has multiple implementation for the different cloud providers, but to make it simpler I only created utility method with general name. I can change the name for now